src/pages/profile/views/badges/Badges.tsx

```diff
@@ -1,93 +1,93 @@
-    if (userBadgeLoading || isLoading) {
-      return <BadgesBodySkeleton />
-    }
-
-    return (
-      <BadgesBody
-        {...{
-          userBadges: userBadgesData?.userBadges || [],
-          userProfile,
-          isEdit,
-        }}
-      />
-    )
-  }
-
-  return (
-    <CardLayout
-      noMobileBorder
-      padding="0"
-      paddingY={{ base: '10px', lg: '20px' }}
-      spacing="20px"
-      maxHeight="100%"
-    >
-      <HStack
-        paddingX={{ base: '10px', lg: '20px' }}
-        w="full"
-        justifyContent="space-between"
-      >
-        <H2>{t('Badges')}</H2>
-        <Button
-          as={ChakraLink}
-          href={getPath('badges')}
-          isExternal
-          size="sm"
-          leftIcon={<BsBoxArrowUpRight />}
-        >
-          {t('See badges')}
-        </Button>
-      </HStack>
-      <Body2 paddingX={{ base: '10px', lg: '20px' }} color="neutral.700">
-        {t(
-          'Geyser badges are earned for launching successful projects, contributing  to them and being an active community member.',
-        )}
-      </Body2>
-
-      {renderBadgesBody()}
-    </CardLayout>
-  )
-}
+import { Button, HStack, Link as ChakraLink } from '@chakra-ui/react'
+import { useEffect } from 'react'
+import { useTranslation } from 'react-i18next'
+import { BsBoxArrowUpRight } from 'react-icons/bs'
+
+import { CardLayout } from '../../../../components/layouts'
+import { Body2, H2 } from '../../../../components/typography'
+import { getPath } from '../../../../constants'
+import { User, useUserBadgesLazyQuery } from '../../../../types'
+import { toInt, useNotification } from '../../../../utils'
+import { BadgesBody, BadgesBodySkeleton } from './BadgesBody'
+
+export const Badges = ({
+  userProfile,
+  isEdit,
+  isLoading,
+}: {
+  userProfile: User
+  isEdit: boolean
+  isLoading: boolean
+}) => {
+  const { t } = useTranslation()
+  const { toast } = useNotification()
+
+  const [getUserBadges, { data: userBadgesData, loading: userBadgeLoading }] =
+    useUserBadgesLazyQuery({
+      onError() {
+        toast({
+          title: 'Error fetching badges',
+          description: 'Please refresh the page and try again.',
+          status: 'error',
+        })
+      },
+    })
+
+  useEffect(() => {
+    if (userProfile.id) {
+      getUserBadges({
+        variables: { input: { where: { userId: toInt(userProfile.id) } } },
+      })
+    }
+  }, [userProfile.id, getUserBadges])
+
+  const renderBadgesBody = () => {
+    if (userBadgeLoading || isLoading) {
+      return <BadgesBodySkeleton />
+    }
+
+    return (
+      <BadgesBody
+        {...{
+          userBadges: userBadgesData?.userBadges || [],
+          userProfile,
+          isEdit,
+        }}
+      />
+    )
+  }
+
+  return (
+    <CardLayout
+      noMobileBorder
+      padding="0"
+      paddingY={{ base: '10px', lg: '0px' }}
+      spacing="20px"
+      style={{ position: 'relative', maxHeight: '320px', overflowY: 'auto' }}

```

Comment:
adding the maxHeight to 320px would work on the desktop mode but for the mobile mode, we'd want it to expand so that there is only one main body scroll.

Should be a small change, instead of using style, pass props directly and use:

maxHeight: {{base: '100%', lg: '320x'}} but I still think there might be other ways to do this, as limiting height is only good for the screen size you used, for larger screen size, expanded content is preferred.

Reply:
it could be because I uploaded a tsx file since I edited it locally that’s why git would read it as such. 

And the scroll section is for desktop view. as mentioned i only used the template from my profile with one badge. once merged, it would be good to see what the layout is like for other users with more badges.(on mobile view) tho I dint change this part , so it should resolve as is on production.